app/src/NativeBase/Components/Project/ProjectRegisterInterest.tsx

```diff
@@ -0,0 +1,221 @@
+/**
+ * @file A form for the user to register interest in taking part in a volunteer project.
+ */
+
+import React, { FC, useEffect, useState } from 'react'
+import { ScrollView, VStack, useColorModeValue, Button } from 'native-base'
+import dayjs from 'dayjs'
+import ButtonComponent from '../Forms/Button'
+import YesNoChoice from '../Forms/YesNoChoice'
+import TextInputControl from '../Forms/TextInputControl'
+import DateTime from '../Forms/DateTime'
+import ResponseModal from '../Forms/ResponseModal'
+import { goBack } from '@/Navigators/utils'
+import {
+  Project,
+  useLazyProjectRegisterInterestQuery,
+} from '@/Services/modules/projects'
+import { validateEmail } from '@/Utils/Validation'
+
+interface ProjectRegisterInterestProps {
+  project: Project
+}
+
+/**
+ * Form for user to register interest in a project
+ *
+ * @param {ProjectRegisterInterestProps} props The component props
+ * @param {Project} props.project The project the user's registering interest in
+ * @returns {React.ReactElement} Component
+ */
+
+const ProjectRegisterInterest: FC<ProjectRegisterInterestProps> = ({
+  project,
+}) => {
+  const [availableFromDate, setAvailableFromDate] = useState(new Date())
+  const [email, setEmail] = useState('')
+  const [errors, setErrors] = useState<{
+    [key: string]: { type: 'invalid' | 'missing' }
+  }>({})
+  const [firstName, setFirstName] = useState('')
+  const [lastName, setLastName] = useState('')
+  const [lookingForBuddy, setLookingForBuddy] = useState(false)
+  const today = new Date()
+  const oneYearInTheFuture = new Date(
+    new Date().setFullYear(today.getFullYear() + 1),
+  )
+  const [registerInterest, { data: responseData, error: responseError }] =
```

Comment:
Good question you raised about the API -- we should remove the mentoring field from being sent to the API and to Slack for now.  Give me a shout if you're not sure how to do this -- hopefully it should just be a case of removing a few lines of code.